File: src/fs/classes.fs
type MontyPython(firstName, lastName) =
    let mutable firstName = firstName
    let mutable lastName = lastName

    member this.FirstName
        with get () = firstName
        and set (value) = firstName <- value

    member this.LastName
        with get () = lastName
        and set (value) = lastName <- value

    member this.FullName =
        sprintf "%s %s" this.FirstName this.LastName

let montyPython = new MontyPython("John", "Cleese")
printfn "Monty Python is %s." montyPython.FullName

montyPython.FirstName <- "Eric"
montyPython.LastName <- "Idle"
printfn "Now, Monty Python is %s." montyPython.FullName

File: src/pl/classes.pl
package MontyPython;
sub new{
    my $class = shift;
    my ($args) = @_;
    my $self = {
        firstName => $args->{firstName},
        lastName => $args->{lastName}
    };

    return bless $self, $class;
}

sub fullName{
    my $self = shift;
    return $self->{firstName} . " " . $self->{lastName};
}

package Main;
$montyPython = MontyPython->new({
    firstName => "John",
    lastName => "Cleese"
});
print "Monty Python is " . $montyPython->fullName() . ".\n";

$montyPython->{firstName} = "Eric";
$montyPython->{lastName} = "Idle";
print "Now, Monty Python is " . $montyPython->fullName() . ".";